### PR TITLE
Feature multiple table

### PR DESCRIPTION
**Pull request by vikipretium**
# [Vet clinic database: query multiple tables](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md#vet-clinic-database-query-multiple-tables)

> * I created a table named `owners` with the following columns:
> * `id`: integer (set it as autoincremented PRIMARY KEY) ✔️ 
> * `full_name`: string ✔️ 
> * `age`: integer ✔️ 
> * I created a table named `species` with the following columns:
>   
>   * `id`: integer (set it as autoincremented PRIMARY KEY) ✔️ 
>   * `name`: string ✔️ 
> * I modified the `animals` table: ✔️ 
>   
>   * I removed column `species` ✔️ 
>   * I added column `species_id` which is a foreign key referencing `species` table ✔️ 
>   * I added column `owner_id` which is a foreign key referencing the `owners` table ✔️ 
> * The following data got inserted into the `owners` table:
>   
>   * Sam Smith 34 years old. ✔️ 
>   * Jennifer Orwell 19 years old. ✔️ 
>   * Bob 45 years old. ✔️ 
>   * Melody Pond 77 years old. ✔️ 
>   * Dean Winchester 14 years old. ✔️ 
>   * Jodie Whittaker 38 years old. ✔️ 
> * The following data got inserted into the `species` table:
>   
>   * Pokemon ✔️ 
>   * Digimon ✔️ 
> * I Modified the inserted animals so it includes the `species_id` value:
>   
>   * If the name ends with "mon"  it will be Digimon ✔️ 
>   * All other animals are Pokemon ✔️ 
> * I modified the inserted animals to include owner information (`owner_id`):
>   
>   * Sam Smith owns Agumon. ✔️ 
>   * Jennifer Orwell owns Gabumon and Pikachu. ✔️ 
>   * Bob owns Devimon and Plantmon. ✔️ 
>   * Melody Pond owns Charmander, Squirtle, and Blossom. ✔️ 
>   * Dean Winchester owns Angemon and Boarmon. ✔️ 
> * I wrote queries (using `JOIN`) to answer the following questions:
>   
>   * What animals belong to Melody Pond? ✔️ 
>   * List of all animals that are pokemon (their type is Pokemon). ✔️ 
>   * List all owners and their animals, remember to include those that don't own any animal. ✔️ 
>   * How many animals are there per species? ✔️ 
>   * List all Digimon owned by Jennifer Orwell. ✔️ 
>   * List all animals owned by Dean Winchester that haven't tried to escape. ✔️ 
>   * Who owns the most animals? ✔️ 

Thanks for your time to review my Pull request 👍🏼

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -10,4 +10,32 @@ CREATE TABLE animals(id INT GENERATED ALWAYS AS IDENTITY,
...
\ No newline at end of file
+ADD COLUMN species varchar(20);
+
+
+CREATE TABLE owners (
+ id INT GENERATED ALWAYS AS IDENTITY,
+ full_name varchar(100) NOT NULL,
+ age SMALLINT NOT NULL,
+ PRIMARY KEY (id)
+);
```
Comment (mateo951) [inline schema.sql:20]: Great! `owners` table was added correctly ✔

File: schema.sql [hunk 2]
```diff
@@ -10,4 +10,32 @@ CREATE TABLE animals(id INT GENERATED ALWAYS AS IDENTITY,
...
+
+CREATE TABLE species (
+ id INT GENERATED ALWAYS AS IDENTITY,
+ name varchar(100) NOT NULL,
+ PRIMARY KEY (id)
+);
```
Comment (mateo951) [inline schema.sql:26]: Great! `species` table was added correctly ✔

File: schema.sql [hunk 3]
```diff
@@ -10,4 +10,32 @@ CREATE TABLE animals(id INT GENERATED ALWAYS AS IDENTITY,
...
+
+ALTER TABLE animal
+DROP COLUMN species;
+
```
Comment (mateo951) [inline schema.sql:30]: Removed column `species` 🔥

File: queries.sql [hunk 4]
```diff
@@ -92,4 +92,58 @@ SET weight_kg = weight_kg * -1;
 WHERE weight_kg < 0 ;
 SELECT * FROM animal;
 
-COMMIT;
\ No newline at end of file
+COMMIT;
+
```
Comment (mateo951) [inline queries.sql]: Please be sure to add the **owners** data into your data.sql file 🧐

```sql
INSERT INTO owners (full_name, age)
 VALUES ('Sam Smith', 34);

 INSERT INTO owners (full_name, age)
 VALUES ('Jennifer Orwell', 19);

 INSERT INTO owners (full_name, age)
 VALUES ('Bob', 45);

 INSERT INTO owners (full_name, age)
 VALUES ('Melody Pond', 77);

 INSERT INTO owners (full_name, age)
 VALUES ('Dean Winchester', 14);

 INSERT INTO owners (full_name, age)
 VALUES ('Jodie Whittaker', 38);
```

Comment (mateo951) [reply queries.sql]: Also kindly add these to species table 

```sql
 INSERT INTO species (name)
 VALUES ('Pokemon');

 INSERT INTO species (name)
 VALUES ('Digimon');
```

File: queries.sql [hunk 5]
```diff
@@ -92,4 +92,58 @@ SET weight_kg = weight_kg * -1;
...
+SET owner_id = 5
+WHERE name IN ('Angemon','Boarmon');
+
+UPDATE animals
+SET species_id = 2
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species_id = 1
+WHERE species_id IS NULL;
```
Comment (mateo951) [inline queries.sql]: Kindly move your UPDATE lines to your schema.sql file 📝 Everything that creates or **changes** the tables structure should be saved in the schema.sql file.

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @vikipretium,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: queries.sql (hunk 4 above)
Comment (vikipretium) [reply queries.sql]: I fixed it @mateo951 👍🏼

File: queries.sql (hunk 5 above)
Comment (vikipretium) [reply queries.sql]: I moved it to schema.sql file @mateo951. Thanks for your time reviewing my PR. It was a great help to me. I understand very well. I added owners and species in my terminal. I forgot to add it to my repository files.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @vikipretium  👋🏻, 

Good job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**